typescript/infra/src/agents/aws/s3.ts

```diff
@@ -0,0 +1,47 @@
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { Readable } from 'stream';
+
+import { utils } from '@abacus-network/utils';
+
+export const S3_BUCKET_REGEX =
+  /^(?:https?:\/\/)?(.*)\.s3\.(.*)\.amazonaws.com\/?$/;
+
+export interface S3Receipt<T = unknown> {
+  data: T;
+  modified: Date;
+}
+
+export class S3Wrapper {
+  private readonly client: S3Client;
+  readonly bucket: string;
+
+  constructor(bucketUrl: string) {
+    const match = bucketUrl.match(S3_BUCKET_REGEX);
+    if (!match) throw new Error('Could not parse bucket url');
+    this.bucket = match[1];
+    const region = match[2];
+    this.client = new S3Client({ region });
+  }
+
+  async getS3Obj<T>(key: string): Promise<S3Receipt<T> | undefined> {
```

Comment:
We might want to start adding docs now that this is getting library-ified. I can do that if you like.

Reply:
I figured we may need to interact with S3 beyond checkpoints and it was simple enough to pull this out